### PR TITLE
Refactor APIBluerprint

### PR DESCRIPTION
**Pull request by shimastripe**
## WHAT
- [x] Modify to resource path.
- [x] Change fixed option position.
- [x] Convert pluralize lowercase to snake_case.
- [x] Convert pluralize lowercase to original_case.
- [x] Add the type to request in model field.
- [x] Bug fix in model field.
- [x] Accept null value.
- [x] Convert real value to example value.

### CODE REVIEW

**Comment by shimastripe** [comment]
All checks have passed. Plz check 🙇

File: apig/generate.go [hunk 1]
```diff
@@ -40,47 +44,70 @@ var managedFields = []string{
...
 	case AssociationHasOne:
-		return inflector.Pluralize(strings.ToLower(strings.Replace(field.Type, "*", "", -1)))
+		return strings.ToLower(strings.Replace(field.Type, "*", "", -1))
 	}
 
 	return ""
 }
 
+func article(s string) string {
+	switch {
```
Comment (dtan4) [inline apig/generate.go]: This compares the first character only, so how about this:

```go
switch s[0] {
case "a", "b", "c", "d", "e":
	return "an " + s
default:
	return "a " + s
}
```

File: apig/generate.go [hunk 2]
```diff
@@ -110,6 +137,26 @@ func camelToLowerCamel(s string) string {
 	return strings.Join(ss, "")
 }
 
+func camelToOriginal(s string) string {
```
Comment (dtan4) [inline apig/generate.go]: What does `Original` mean?

File: apig/generate.go (hunk 1 above)
Comment (shimastripe) [inline apig/generate.go]: OK, I fixed it! 👍

File: apig/generate.go (hunk 2 above)
Comment (shimastripe) [inline apig/generate.go]: It is a compound word of space-separated.

original: `account name`
lowerCamel: `accountName`
snake_case: `account_name`

Comment (dtan4) [inline apig/generate.go]: OK. please add comment above function.

`// accountName -> account name`

Comment (shimastripe) [inline apig/generate.go]: Thx. I inserted a comment.

**Comment by dtan4** [comment]
LGTM 👍 Thanks ❗

**Pull request merged**